core/chains/evm/txmgr/eth_confirmer.go

```diff
@@ -1032,8 +1031,8 @@ func (ec *EthConfirmer) markForRebroadcast(etx EthTx, head *evmtypes.Head) error
 
 	ec.lggr.Infow(fmt.Sprintf("Re-org detected. Rebroadcasting transaction %s which may have been re-org'd out of the main chain", attempt.Hash.Hex()),
 		"txhash", attempt.Hash.Hex(),
-		"currentBlockNum", head.Number,
-		"currentBlockHash", head.Hash.Hex(),
+		"currentBlockNum", head.BlockNumber(),
+		"currentBlockHash", head.BlockHash().Hex(),
```

Comment:
note: example of the blockhash being bound to the geth implementation

Reply:
Yes, we can't use a generic type for hash because of this. We will instead need a Hash interface, which has common methods, like ToString(), Compare(), Serialize(), DeSerialize().

And that's why I didn't want to do that in this PR. Trying to keep changes isolated and small, for easier reviews.